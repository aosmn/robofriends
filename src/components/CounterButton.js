import React, { Component } from 'react'

export default class CounterButton extends Component {
  constructor(props) {
    super(props);
    
    this.state = {
      count: 0
    }
  }
  
  render() {
    return (
      <button color={this.props.color}
      id='counter'
      onClick={() => this.setState(state => ({ count: state.count+1 }))} >
      Count: {this.state.count}
      </button>
    )
  }
}
