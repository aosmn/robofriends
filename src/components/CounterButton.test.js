import { shallow} from 'enzyme';
import React from 'react';
import CounterButton from './CounterButton';

it('should render cardList component', () => {
  const mockColor = 'red';
  expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot()
});

it('should correctly increment the counter', () => {
  const mockColor = 'red';
  const wrapper = shallow(<CounterButton color={mockColor} />);

  wrapper.find('[id="counter"]').simulate('click');
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 2 })
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 3 })
  expect(wrapper.props().color).toEqual('red')
});