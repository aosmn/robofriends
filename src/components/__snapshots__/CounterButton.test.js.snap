// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render cardList component 1`] = `ShallowWrapper {}`;
